src/fs.h

```diff
@@ -103,4 +226,11 @@ namespace fsbridge {
 #endif // WIN32 && __GLIBCXX__
 };
 
+// Disallow implicit std::string conversion for tinyformat to avoid
+// locale-dependent encoding on windows.
+namespace tinyformat {
+template<> inline void formatValue(std::ostream&, const char*, const char*, int, const boost::filesystem::path&);
+template<> inline void formatValue(std::ostream&, const char*, const char*, int, const fs::path&);
+} // namespace tinyformat
```

Comment:
A note (not an issue).
If code contains a `formatValue` call which assumes implicit `std::string` conversion, an error will be generated by a _linker_, not a compiler.

Reply:
```suggestion
// Disallow implicit std::string conversion for tinyformat to avoid
// locale-dependent encoding on windows.
namespace tinyformat {
template<> inline void formatValue(std::ostream&, const char*, const char*, int, const fs::path&) = delete;
} // namespace tinyformat
```

to force a compiler to report errors?

Also the explicit template specialization with the `boost::filesystem::path` type seems redundant, no?

Reply:
> to force a compiler to report errors?

Wow, I had no idea you could delete nonmember functions. It is much nicer to have this caught by the compiler. Will update.

> Also the explicit template specialization with the `boost::filesystem::path` type seems redundant, no?

This is just to things cases like `strformat("%s", path1 / path2)` or whatever else might return a `std::filesystem::path` instead of an `fs::path`

Reply:
> This is just to catch cases like `strformat("%s", path1 / path2)` or whatever else might return a `std::filesystem::path` instead of an `fs::path`

We do not use the same assumption for other functions. Why just here?

Reply:
> We do not use the same assumption for other functions. Why just here?

If there are other places where it's easy to catch more encoding problems, I would be happy to add overloads. This case does work a little differently from other cases because it is template specialization not a function overload. A function overload will match when a subclass instance is passed to a parent class parameter, while template specialization will not match because type deduction won't deduce a parent type from a child type.

Reply:
Thanks for an explanation!

Reply:
If we did want to be more vigilant about possible encoding bugs, we could have a followup PR that drops public inheritance `class path : public boost::filesystem::path` and wraps a private member instead, or uses private inheritance, to make the class a whitelist of allowed functions instead of a blacklist of disallowed functions. The changes required would be a superset of changes already in this PR, and it's unclear whether any bugs this might catch would be severe enough to justify the extra maintenance burden, so I didn't try this here. But it is something that could be done later.